hal/src/rtl872x/interrupts_hal.cpp

```diff
@@ -212,9 +317,16 @@ int hal_interrupt_attach(uint16_t pin, hal_interrupt_handler_t handler, void* da
         GPIO_UserRegIrq(rtlPin, (VOID*)gpioIntHandler, (void*)((uint32_t)pin));
         GPIO_INTMode_HAL(rtlPin, pinInfo->gpio_port, ENABLE, GPIO_InitStruct.GPIO_ITTrigger, GPIO_InitStruct.GPIO_ITPolarity, GPIO_INT_DEBOUNCE_ENABLE);
 
-        interruptsConfig[pin].state = INT_STATE_ENABLED;
+#if MODULE_FUNCTION != MOD_FUNC_BOOTLOADER
+        if (!config || (config && !(config->appendHandler))) {
+            interruptsConfig[pin].clearHandlers();
+        }
+        interruptsConfig[pin].appendHandler(handler, data, config->chainPriority);
```

Comment:
Ideally `config->chainPriority` can only be accessed if there is such a field, which as of right now indicated by `config->version == HAL_INTERRUPT_EXTRA_CONFIGURATION_VERSION_3`. Presence of `config->appendHandler` does not necessarily mean that we are not going to access out of bounds.